views/pages/RegisterPage.jsx

```diff
@@ -1,280 +1,118 @@
-import CardContent from '@material-ui/core/CardContent'
-import CardMedia from '@material-ui/core/CardMedia'
-import Typography from '@material-ui/core/Typography'
-import TextField from '@material-ui/core/TextField'
-import Button from '@material-ui/core/Button'
-import { withRouter } from '../hoc/withRouter'
-
-import basePathConfig from '../../server/configs/basePathConfig'
-
-const basePath = basePathConfig || ''
-
-class RegisterPage extends Component {
-  constructor(props) {
-    super(props)
-    this.state = {
-      message: '',
+const RegisterPage = ({ classes }) => {
+  const [width, setWidth] = useState(window.innerWidth)
+  const [errors, setErrors] = useState({
+    password: { error: false, message: '' },
+    email: { error: false, message: '' },
+    username: { error: false, message: '' },
+  })
+  const { handleSubmit, control, reset } = useForm({
+    defaultValues: {
       username: '',
       password: '',
+      confirmPassword: '',
       email: '',
-      fullname: '',
-      confirmpw: '',
-      unmatched: false,
-      unInitialized: false,
-      fnInitialized: false,
-    }
-  }
-  componentDidUpdate() {}
-  handleClick = () => {
-    this.setState({
-      open: true,
-    })
-  }
-  handleRequestClose = () => {
-    this.setState({
-      open: false,
-    })
-  }
-  handleChange = (prop) => (event) => {
-    if (prop === 'fullname' && !this.state.fnInitialized) {
-      this.setState({
-        [prop]: event.target.value,
-        fnInitialized: true,
-      })
-      return
-    } else if (prop === 'username' && !this.state.unInitialized) {
-      this.setState({
-        [prop]: event.target.value,
-        unInitialized: true,
-      })
-      return
+      fullName: '',
+    },
+  })
+  const [, setNotification] = useContext(NotificationContext)
+  const navigate = useNavigate()
+  const handleFormSubmit = async (data) => {
+    switch (true) {
+      case data.password !== data.confirmPassword:
+        setErrors({
+          ...errors,
+          password: { error: true, message: 'passwords do not match' },
+        })
+
+        break
+      case data.password.length < 8 || data.confirmPassword.length < 8:
+        setErrors({
+          ...errors,
+          password: {
+            error: true,
+            message: 'passwords must be a minimum of 8 characters',
+          },
+        })
+
+        break
+      case !VALIDATION_EMAIL_REGEX.test(data.email):
+        setErrors({
+          ...errors,
+          email: { error: true, message: 'incorrect email format' },
+        })
+
+        break
+      default:
+        setErrors({
+          password: { error: false, message: '' },
+          email: { error: false, message: '' },
+        })
+
+        await api.auth.signup(data)
+
+        setNotification({
+          open: true,
+          message:
+            'Account has been created, please wait for an Admin to provide access.',
+        })
+
+        reset()
```

Comment:
do we want to reset the form? I think this will remove any changes the user made (including valid ones), vs leaving the form as-is and allowing the user to update the invalid fields.

Reply:
We only reset the form after it's successfully submitted, there's a notification popup that alerts the user that they will receive access when an admin permits them. Unfortunately, they don't get access right away.

Reply:
[c8a966a](https://github.com/AMP-SCZ/dpdash/pull/394/commits/c8a966a72451379d6eb5324fa0a3279b56ff48f5)